src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/IGDBClientTests.kt

```diff
@@ -1,4 +1,169 @@
 package com.gamingbacklog.api.gamingbacklogapi.clients
 
+import com.gamingbacklog.api.gamingbacklogapi.testutil.models.MockResponse
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.given
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.TestPropertySource
+
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+
+@SpringBootTest
+@Import(IGDBClient::class)
+@TestPropertySource(properties = ["CLIENT_ID=test_id", "CLIENT_SECRET=test_secret"])
 class IGDBClientTests {
+  lateinit var externalAPICall: ExternalAPICall
+  lateinit var igdbClient: IGDBClient
+
+  @BeforeEach
+  fun configureSystem() {
+    externalAPICall = mock(ExternalAPICall::class.java)
+    igdbClient = IGDBClient(externalAPICall)
+  }
+
+  @Nested
+  @DisplayName("Tests for Authenticate")
+  inner class Authenticate {
+
+    @Test
+    fun shouldSuccessfullyReturnToken() {
+      given(externalAPICall.postExternalCall(any(), eq(null), eq(null))).willReturn(
+        MockResponse(
+          "{\"access_token\":\"test_secret\",\"expires_in\":5180593,\"token_type\":\"bearer\"}",
```

Comment:
Do we want to define a class for the expected returns rather than just leave it as a string? (Can do in a separate ticket)